services/src/main/java/org/fao/geonet/api/site/SitemapApi.java

```diff
@@ -209,6 +210,7 @@ ResponseEntity<Element> getSitemap(
         root.addContent(result);
 
         Element sitemap = Xml.transform(root, xslt);
-        return new ResponseEntity<>(sitemap, HttpStatus.OK);
+        response.setHeader(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_XML_VALUE);
+        response.getWriter().append(Xml.getString(sitemap));
```

Comment:
Maybe we could use https://github.com/geonetwork/core-geonetwork/blob/main/common/src/main/java/org/fao/geonet/utils/Xml.java#L409 directly with `response.getOutputStream()` ?

Reply:
Changed as suggested.